corehq/apps/reports/filters/users.py

```diff
@@ -295,9 +296,13 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs):
             return q.OR(*user_type_filters)
         else:
             # return matching user types and exact matches
+            location_ids = list(SQLLocation.active_objects
+                                .get_locations_and_children(location_ids)
+                                .location_ids())
             id_filter = filters.OR(
                 filters.term("_id", user_ids),
                 filters.term("__group_ids", group_ids),
+                user_es.location(location_ids),
```

Comment:
offlined with @benrudolph and we convinced ourselves that this is better behavior.